edx_proctoring/views.py

```diff
@@ -489,6 +494,65 @@ def get(self, request):
         elif relevant_attempt:
             data['onboarding_status'] = relevant_attempt.status
 
+        if waffle.switch_is_active(ONBOARDING_STATUS_API):
+            attempt_data = None
+            try:
+                attempt_data = backend.get_onboarding_attempts(course_id=course_id, user_id=user.id)
+            except BackendProviderOnboardingStatusesException as exc:
+                log_message = (
+                    'Failed to use backend onboarding status API endpoint for user_id={user_id}'
+                    'in course_id={course_id} because backend failed to respond. '
+                    'Status: {status}, Response: {response}'.format(
+                        user_id=user.id,
+                        course_id=course_id,
+                        response=str(exc),
+                        status=exc.http_status,
+                    )
+                )
+                LOG.warning(log_message)
+                return Response(data)
+
+            if attempt_data is None:
+                return Response(
+                    data = _('No onboarding status API for {proctor_service}').format(
+                        proctor_service=backend.verbose_name
+                    ),
+                    status = 404,
+                )
+            else:
+                readable_status = \
+                    VerificientOnboardingAttemptStatus.get_attempt_status_from_api_onboarding_status(
+                        attempt_data['status']
+                    )
+                expiration_date = attempt_data['expiration_date']
+
+                if readable_status != data['onboarding_status'] or expiration_date != data['expiration_date']:
```

Comment:
If the data from the endpoint differs from our logic, which data should we choose to use?